### ISSUE TITLE
30% of DB objects have an inconsistent local path

### ISSUE THREAD

**Issue by grololo06** [ecotaxa/ecotaxa_dev#541]
Hello,
In mind, we have for each object, a path project -> sample -> acquisition -> process with 1,n cardinality on each arrow.
It's never been enforced on code side.
Let's check if this is true, and if not how many problems exist. For that, I created a dedicated table on production DB:
```
ecotaxa=# create table obj_path as select projid, sampleid, acquisid, processid, count(objid) as cnt 
          from obj_head 
          group by projid, sampleid, acquisid, processid;
SELECT 167517
ecotaxa=# grant select on obj_path to zoo;
GRANT
```
Check the count:
```
select sum(cnt)  from obj_path op 			  
union all
select count(1) from obj_head
131894350
131894350
```

So, according to the "rule", if objects are in the same process, they should be in the same path. Let's check:

```
select sum(cnt) 
  from obj_path op 
where exists (select 1 
		    from obj_path op2 
		   where op2.processid = op.processid   -- same process
			 and (op2.projid != op.projid     -- but different something else
				  or op2.sampleid != op.sampleid
				  or op2.acquisid != op.acquisid))
**43072283**
```

**Comment by grololo06** [comment]
Down to 19% after upgrade script of #367

**Comment by grololo06** [comment]
Update 14/01/2021
```
create temp table obj_path as select projid, sampleid, acquisid, count(objid) as cnt 
          from obj_head 
          group by projid, sampleid, acquisid;
          
select sum(cnt) 
  from obj_path op 
where exists (select 1 
		    from obj_path op2 
		   where op2.acquisid = op.acquisid   -- same acquisition
			 and (op2.projid != op.projid     -- but different something else
			  or op2.sampleid != op.sampleid))  
```
25584365

**Comment by grololo06** [comment]
Linked with #544

**Comment by grololo06** [comment]
Fixed in 2.5.2
